examples/wasi-http-rust-streaming-outgoing-body/src/lib.rs

```diff
@@ -83,18 +89,121 @@ async fn handle_request(request: IncomingRequest, response_out: ResponseOutparam
             }
         }
 
-        _ => {
-            let response = OutgoingResponse::new(405, &Headers::new(&[]));
+        (Method::Post, Some("/double-echo")) => {
+            // Pipe the request body to an outgoing request and stream the response back to the client.
 
-            let body = response.write().expect("response should be writable");
-
-            response_out.set(response);
+            if let Some(url) = headers.iter().find_map(|(k, v)| {
+                (k == "url")
+                    .then_some(v)
+                    .and_then(|v| std::str::from_utf8(v).ok())
+                    .and_then(|v| Url::parse(v).ok())
+            }) {
+                match double_echo(request, &url).await {
+                    Ok((request_copy, response)) => {
+                        let mut stream = response.take_body_stream();
```

Comment:
Finding it a bit hard to follow with the generic naming: perhaps using slightly more verbose names might help?

```suggestion
                        let mut incoming_response_body = response.take_body_stream();
```